### DISCUSSION TITLE
Non-docker install and update

### DISCUSSION THREAD

**Discussion by nicoleise** [blakeblackshear/frigate#6545] [Ask A Question] [answered]
Hi, 

Due to the issues with Proxmox, Docker and ZFS in LXCs (not specific to Frigate), I'd like to avoid using Docker. 

I have tested, and I get the usual issue with this combination and Frigate (containers and images disappear from Docker after reboot or restore from backup of guest, but files are still in /var/ folder).

I've also read the discussion thread here asking for a bare metal install option. While I would love it, I'm not asking for that, and realize it'd be a big change for the project. 

What I am wondering, however... Previously I have simply recreated the dockerfile manually (for other projects), by starting with the target OS and applying the commands the file contains. This has worked fine for my needs, since it essentially ends with the same result.

If I do the same with Frigate, likely updating will be an issue, since Frigate is very much actively maintained, whereas the projects I've previously used this approach on were either dead or short term uses on my part.

Any pointers as to how I can achieve updates to Frigate with a bare metal style install? Is there a complete change log I could apply (step by step commands) or an update mechanism in Frigate itself?

And also, can Frigate be installed on newer OS versions than the version declared in the dockerfile (Debian 10), or are there underlying dependencies that will not work in fx Debian 12?

Kind regards, and thanks for a great project,
Nicolai

**Comment by blakeblackshear** [comment, marked as answer]
You just have to run the commands in the Dockerfile. Maybe there is some kind of utility that will translate a Dockerfile into a bash script. I don't know if it will work on newer versions of Debian than what is in the Dockerfile. I would be prepared to reinstall your baremetal OS from scratch for each release.

To clarify, you are running Proxmox? Where do you intend to run the commands in the Dockerfile?

**Reply by nicoleise** [reply]
>  I would be prepared to reinstall your baremetal OS from scratch for each release.

Right, so essentially no update path with this method? (I suspected as much, especially ffmpeg seems super-finnicky about versions and OS versions - just wanting to confirm).

> To clarify, you are running Proxmox? Where do you intend to run the commands in the Dockerfile?

Yes. What I've previously done is to stand up a LXC with the OS defined in the dockerfile (from target) and then manually executing each command. While tedious, it has worked without issue and avoids the docker/LXC issues. However with Frigate, updating seems much more important than with the projects I've previously used this approach on. So having a somewhat smooth upgrade path would be nice.

**Reply by blakeblackshear** [reply]
Have you considered something like this? https://bobcares.com/blog/convert-docker-image-to-lxd/

**Reply by nicoleise** [reply]
> Have you considered something like this? https://bobcares.com/blog/convert-docker-image-to-lxd/

Yes, see #5448 for added detail. For now, I have setup a VM with Docker and installed via Docker Compose. Thanks. :)
